File: scripts/maui-vml-verify/corpus/pascal/skel.pas
{ skel.pas —— VML 骨架程序（Pascal）。

  习惯用法抄自 Examples/pascal/stm32/f103/stm32f103_spi.pas（program + begin/end.，
  裸调库函数不写声明）；array / for-to / function 见 docs/前端游戏能力评估.md §4.2'
  （patch 0013 修掉 L0 标签与数组地址方向后，Pascal 四关全过、心跳也验过）。
  颜色写负数十进制：本前端没有一处 $ 或 0x 的现成示例，不冒这个险。
  函数名故意不叫 inc —— Pascal 标准过程里就有 Inc()。
  WriteLn 支持多实参、实参之间不插分隔符、末尾补一次换行 ⇒ 直接拼出一行。 }
program Skel;

var
  a: array[0..3] of integer;
  i: integer;
  s: integer;

function plus1(x: integer): integer;
begin
  plus1 := x + 1;
end;

begin
  a[0] := 1;
  a[1] := 2;
  a[2] := 3;
  a[3] := 4;
  s := 0;
  for i := 0 to 3 do
  begin
    a[i] := plus1(a[i]);
    s := s + a[i];
  end;
  ui_rect(10, 10, 50, 50, -65536, 1, 0, 0);
  ui_present();
  WriteLn('SKEL-SUM=', s);
end.

File: scripts/maui-vml-verify/corpus/lua/skel.lua
-- skel.lua —— Lua 前端「能不能写游戏」最小骨架（期望输出恰好一行 SKEL-SUM=14）
--
-- 取材 Examples/lua/stm32/f103/stm32f103_i2c.lua（`function main() ... end` + 顶层 `main()`）；
-- 前端实现 VMLPrepares/LuaCompiler/。
--
-- 故意踩的已知缺陷（跑不过=产品缺陷，不是语料写错）：
--  ① `a[i]` / `a[i] = v` 是表索引，前端编成 `CALL lua_table_get` / `CALL lua_table_set`，
--     而这两个标签在 Lib/ 下**根本不存在**（CodeGenerator.Statements_B.cs:1034、
--     Statements_A.cs:206）⇒ 链接只报「未解析标签」警告，运行期抛 未找到标签。
--  ② 函数调用只有前 4 个实参进 R0-R3 且**不压栈**（Statements_B.cs:443-471），
--     ui_rect 的 8 个实参到不了包装函数读的 [R12+12]…[R12+40]。
--  ③ `0x…` 字面量在 Parser.cs:662 会 Int32 溢出，颜色写 -65536（= 0xFFFF0000）。
--  ④ print 多实参不加分隔符、末尾自动补换行（Statements_B.cs:476-502）。
--
-- ⚠ 下标基准：Lua 的表是 **1-based**（真 Lua 如此，本前端亦如此 ——
--    `LUA_LANGUAGE_SPEC.md` 明写 `local arr = {10,20,30,40}; print(arr[1]) -- 10（Lua索引从1开始）`，
--    `GenerateTableConstructor` 给位置字段的键也是 `slotIndex + 1`）。
--    本文件原先写的是 `for i = 0, 3` —— **那是语料作者的笔误**（他按 0-based 语言的习惯套过来了；
--    对照 Fortran(1-based) 写的是 `a(1)…a(4)`，Pascal/BASIC/JS/Python(0-based) 写的是 `0..3`）。
--    0-based 取 `a[0..3]` 时 `a[0]` 是新键（nil→0），`s` 只得 10。
--    **修的是语料，不是前端** —— 把前端改成 0-based 会同时推翻真 Lua 语义与它自己的规格文档。

function inc(x)
    return x + 1
end

function main()
    local a = {1, 2, 3, 4}
    local s = 0
    local i = 0
    for i = 1, 4 do
        a[i] = inc(a[i])
        s = s + a[i]
    end
    print("SKEL-SUM=", s)
    ui_rect(10, 10, 50, 50, -65536, 1, 0, 0)
    ui_present()
end

main()
